docusaurus/docs/tutorials/build-a-streaming-data-source-plugin.md

```diff
@@ -15,143 +14,321 @@ keywords:
+- **`data`** - used to exchange information between the frontend and the backend
 
-1. Use `subscriber.next()` to send the updated data frame whenever you receive new updates.
+It's worth mentioning that the `path` can be used to exchange data with the backend part as well, but for complex formats, use the `data` property.
 
-   ```ts
-   import { LoadingState } from '@grafana/data';
-   ```
+:::note
 
-   ```ts
-   const intervalId = setInterval(() => {
-     frame.add({ time: Date.now(), value: Math.random() });
+The stream is uniquely identified by a combination of parameters and it will have the format: `${scope}/${namespace}/${path}`. This is only an implementation detail and you probably won't need to know this except for debugging.
 
-     subscriber.next({
-       data: [frame],
-       key: query.refId,
-       state: LoadingState.Streaming,
-     });
-   }, 500);
+:::
 
-   return () => {
-     clearInterval(intervalId);
-   };
-   ```
+## Step 3: Set up the plugin backend
 
-   :::note
-   In practice, you'd call `subscriber.next` as soon as you receive new data from a websocket or a message bus. In the example above, data is being received every 500 milliseconds.
-   :::
+Now we need to add the necessary code to the backend. For that, we change `pkg/plugin/datasource.go`, this is where the backend part responsible for handling the queries created by the frontend is defined. In our case, as we want to handle a stream, we need to implement the `backend.StreamHandler`.
 
-### Example code for final `query` method
+1. Add the following part in the replacement of the `var`:
 
-```ts
-query(options: DataQueryRequest<MyQuery>): Observable<DataQueryResponse> {
-  const streams = options.targets.map(target => {
-    const query = defaults(target, defaultQuery);
+```tsx
+var (
+    _ backend.CheckHealthHandler    = (*Datasource)(nil)
+    _ instancemgmt.InstanceDisposer = (*Datasource)(nil)
+    _ backend.StreamHandler         = (*Datasource)(nil)
+)
+```
 
-    return new Observable<DataQueryResponse>(subscriber => {
-      const frame = new CircularDataFrame({
-        append: 'tail',
-        capacity: 1000,
-      });
+That means our `DataSource` will implement `backend.CheckHealthHandler`, `instancemgmt.InstanceDisposer`, and `backend.StreamHandler`. The methods necessary for the first two are already provided in the scaffold; therefore, we need to implement only the methods necessary for `backend.StreamHandler`.
 
-      frame.refId = query.refId;
-      frame.addField({ name: 'time', type: FieldType.time });
-      frame.addField({ name: 'value', type: FieldType.number });
+2. This code will be called when the user tries to subscribe to a channel. You can implement permissions checking here, but in our case we just want the user to connect successfully in each attempt; therefore, we just return a backend: `SubscribeStreamStatusOK`.
```

Comment:
From the text content, it seems like the `SubscribeStream` code should be added before this item

Reply:
I'm not quite sure how to implement your suggestion. It sounds like you are saying perhaps that we need to insert a new step between steps 1 and 2 on "Step 3", one which would look like this:

2. Implement code that returns `SubscribeStreamStatusOK` when the user attempts to connect to the backend:

 ```tsx
func (d *Datasource) SubscribeStream(context.Context, *backend.SubscribeStreamRequest) (*backend.SubscribeStreamResponse, error) {
    return &backend.SubscribeStreamResponse{
        Status: backend.SubscribeStreamStatusOK,
    }, nil
}
```

Is that right?